app/models/thredded/messageboard.rb

```diff
@@ -50,11 +56,35 @@ class Messageboard < ActiveRecord::Base
 
     has_many :post_moderation_records, inverse_of: :messageboard, dependent: :delete_all
 
-    default_scope { where(closed: false).order(topics_count: :desc) }
-
+    default_scope { where(closed: false) }
+    # rubocop:disable Style/Lambda
     scope :top_level_messageboards, -> { where(group: nil) }
     scope :by_messageboard_group, ->(group) { where(group: group.id) }
-
+    scope :ordered, ->(order = Thredded.messageboards_order) {
+      case order
+      when :position
+        ordered_by_position
+      when :created_at_asc
+        ordered_by_created_at_asc
+      when :last_post_at_desc
+        ordered_by_last_post_at_desc
+      when :topics_count_desc
+        ordered_by_topics_count_desc
+      end.order(id: :asc)
+    }
+    scope :ordered_by_position, ->() { order(position: :asc) }
+    scope :ordered_by_created_at_asc, ->() { order(position: :asc) }
+    scope :ordered_by_last_post_at_desc, ->() {
+      joins('LEFT JOIN thredded_topics AS last_topics ON thredded_messageboards.last_topic_id = last_topics.id')
+        .order('COALESCE(last_topics.last_post_at, thredded_messageboards.created_at) DESC')
+    }
+    scope :ordered_by_topics_count_desc, ->() {
+      order(topics_count: :desc)
+    }
+    scope :ordered_by_group, ->(order = Thredded.messageboards_order) {
+      includes(:group).order('thredded_messageboard_groups.position asc').ordered(order)
```

Comment:
Perhaps this should be a concerned of `Thredded::MessageboardGroup`, so that here this would look like:

```ruby
includes(:group).merge(Thredded::MessageboardGroup.ordered).ordered(order)
```

And then maybe this scope can be removed (inlined), as it doesn't seem like it's ever useful to order by the group without then grouping by it.